catalog/to-consul/resource.go

```diff
@@ -643,6 +646,17 @@ func (t *ServiceResource) registerServiceInstance(
 			r.Service.ID = serviceID(r.Service.Service, addr)
 			r.Service.Address = addr
 			r.Service.Port = epPort
+			r.Service.Meta = make(map[string]string)
+			for k, v := range baseService.Meta {
+				r.Service.Meta[k] = v
+			}
```

Comment:
Consul metadata has some specific rules that must be followed so we can't map directly:

> The meta object is a map of max 64 key/values with string semantics. Key can contain only ASCII chars and no special characters (A-Z a-z 0-9 _ and -). For performance and security reasons, values as well as keys are limited to 128 characters for keys, 512 for values. This object has the same limitations as the node meta object in node definition. All those meta data can be retrieved individually per instance of the service and all the instances of a given service have their own copy of it.

From https://www.consul.io/docs/discovery/services